python/lsst/obs/decam/decamMapper.py

```diff
@@ -51,17 +51,33 @@ def __init__(self, inputPolicy=None, **kwargs):
 
         super(DecamMapper, self).__init__(policy, policyFile.getRepositoryPath(), **kwargs)
 
+        # Ensure each dataset type of interest knows all useful dataId keys
```

Comment:
The commit message for this (e60b1e4) just refers to DM-5902. On that ticket, @ktlim seems to express some scepticism ("Why should there be a need for all keys to be provided for all dataset types?", etc). Without thinking more than I've done so far, I don't know whether this is necessary, but, even if it is, I do think it needs more explanation and justification.